Make.inc

```diff
@@ -171,7 +171,47 @@ endif
 # Whether to use GPL libraries or not.
 USE_GPL_LIBS ?= 1
 
+# Whether to install Julia as a framework on Darwin (Apple) platforms.
+DARWIN_FRAMEWORK ?= 0
+
+# Override in Make.inc to customize the framework:
+framework_name ?= Julia
+framework_bundleidentifier_base ?= org.julialang.julia
+framework_version ?= $(SOMAJOR)
+
+# Directories and file structure of a Darwin framework:
+framework_bundleidentifier=$(framework_bundleidentifier_base).lib
+framework_directory:=$(framework_name).framework
+framework_versions:=$(framework_directory)/Versions
+framework_currver:=$(framework_versions)/$(framework_version)
+framework_dylib:=$(framework_currver)/$(framework_name)
+framework_headers:=$(framework_currver)/Headers
+framework_documentation:=$(framework_currver)/Documentation
+framework_resources:=$(framework_currver)/Resources
+framework_frameworks:=$(framework_currver)/Frameworks
+framework_infoplist:=$(framework_resources)/Info.plist
+
+# The codesigning identity.
+# used with `codesign -s $(apple_codesign_keychain_identity) $file`
+apple_codesign_keychain_identity := -
+
+# By default, derive the codesign identifiers from the framework bundle id
+apple_codesign_id_julia_ui = $(framework_bundleidentifier_base)
+apple_codesign_id_julia_deps = $(framework_bundleidentifier_base).deps
```

Comment:
I think many if not all of these `framework_*` and `apple_codesign_*` variables could be under contrib/mac.  No problem.